### PR TITLE
Remove unnecessary CopyToDevice + small fixes

### PR DESCRIPTION
**Pull request by ariostas**
I addressed the following comments:

- https://github.com/cms-sw/cmssw/pull/45117#discussion_r1833227743 (move include)
- https://github.com/cms-sw/cmssw/pull/45117#discussion_r1833245889 (unnecessary CopyToDevice specialization)
- https://github.com/cms-sw/cmssw/pull/45117#discussion_r1833246174 (unnecessary CopyToDevice specialization)
- https://github.com/cms-sw/cmssw/pull/45117#discussion_r1833249693 (unused data member)
- https://github.com/cms-sw/cmssw/pull/45117#discussion_r1833298923 (wrong return type)

The CI currently fails because there is a typo in the PortableMultiCollection CopyToDevice definition, which will be fixed in https://github.com/cms-sw/cmssw/pull/46636

### CODE REVIEW

File: RecoTracker/LSTCore/src/alpaka/MiniDoublet.h [hunk 1]
```diff
@@ -85,7 +85,7 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::lst {
     mds.outerLowEdgeY()[idx] = hits.lowEdgeYs()[outerHitIndex];
   }
 
-  ALPAKA_FN_ACC ALPAKA_FN_INLINE float isTighterTiltedModules(ModulesConst modules, uint16_t moduleIndex) {
+  ALPAKA_FN_ACC ALPAKA_FN_INLINE bool isTighterTiltedModules(ModulesConst modules, uint16_t moduleIndex) {
```
Comment (slava77) [inline RecoTracker/LSTCore/src/alpaka/MiniDoublet.h:88]: this is incomplete; cf https://github.com/SegmentLinking/cmssw/pull/115/commits/fc792fd89dd4716c05f44c0fe9e2fdf21891e8db
I'm not sure if identical change is a conflict, hopefully not

**Comment by slava77** [comment]
I'm not a particular expert in the code (`CopyToDevice` ) that was changed.
Otherwise the PR looks OK.
In the past I relied on CI to confirm that things run.
IIUC, the CI will break after this change, right?

**Comment by VourMa** [comment]
> In the past I relied on CI to confirm that things run.
IIUC, the CI will break after this change, right?

I understand that the fix is very simple. Shall we add it to this PR, run the tests and then revert?

**Comment by ariostas** [comment]
> IIUC, the CI will break after this change, right?

yeah, it will break after this change

> I understand that the fix is very simple. Shall we add it to this PR, run the tests and then revert?

That's a good idea. I'll add an extra commit and then remove it once the CI runs

**Comment by ariostas** [comment]
I'm marking it as a draft so we don't forget to remove the temporary commit before merging

**Comment by ariostas** [comment]
I removed the temporary commit, so now this is ready to go

**Review by slava77 — APPROVED** [review]

File: RecoTracker/LSTCore/interface/LSTESData.h [hunk 2]
```diff
@@ -46,41 +46,25 @@ namespace lst {
...
-                  queue, *srcData.modules));
-      auto deviceEndcapGeometry = std::make_unique<PortableCollection<lst::EndcapGeometryDevSoA, alpaka::Dev<TQueue>>>(
-          CopyToDevice<PortableHostCollection<lst::EndcapGeometryDevSoA>>::copyAsync(queue, *srcData.endcapGeometry));
+      using TDev = alpaka::Dev<TQueue>;
+      std::shared_ptr<const PortableMultiCollection<TDev, lst::ModulesSoA, lst::ModulesPixelSoA>> deviceModules;
+      std::shared_ptr<const PortableCollection<lst::EndcapGeometryDevSoA, TDev>> deviceEndcapGeometry;
+
+      if constexpr (std::is_same_v<TDev, alpaka_common::DevHost>) {
+        deviceModules = srcData.modules;
+        deviceEndcapGeometry = srcData.endcapGeometry;
```
Comment (makortel) [inline RecoTracker/LSTCore/interface/LSTESData.h:60]: (for a follow-up to https://github.com/cms-sw/cmssw/pull/45117)

This branch should not be needed, as the `CopyToDevice` should not be called for the CPU backend.

**Pull request merged**